### ISSUE TITLE
🚫🐛 Cannot differentiate between canceled and finished Activities

### ISSUE THREAD

**Issue by ElRaptorus** [process-engine/process_engine_runtime#244]
#### Describe your issue

Resuming BoundaryEvent handlers is currently somewhat twitchy, because it is not possible to tell from their TokenHistory if the BoundaryEvent was triggered or prematurely canceled.

This is due to the fact that BoundaryEvents can only have a `running` and `finished` state.

What we are missing here is an `interrupted` state, which we could apply when the BoundaryEvent was prematurely canceled. 
That will always happen, when the decorated handler has finished, before the BoundaryEvent was triggered.

For non-interrupting BoundaryEvents, which were triggered at least once, the `finished` state will always apply.

Thus, this issue is currently blocked by #235 

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

**Comment by ElRaptorus** [comment]
Effectively a duplicate of #235
